ddtrace/internal/telemetry/writer.py

```diff
@@ -508,6 +511,33 @@ def _app_integrations_changed_event(self, integrations):
         }
         self.add_event(payload, "app-integrations-change")
 
+    def _app_client_configuration_changed_event(self, configuration):
+        # type: (List[Dict]) -> None
+        """Adds a Telemetry event which sends list of modified configurations to the agent"""
+
+        config_change_queue = []
+        # Loop though the events queue and find the app-started event payload
+        for e in self._events_queue:
+            if e["request_type"] == "app-started":
+                payload = e["payload"]
```

Comment:
It looks like `payload` will be undefined if this condition is never hit. I'd recommend adjusting this function's logic to avoid crashing in that situation, and validating that it works with another test.